docs/devel: Specify that `GADGET_*` macros are optional (#2498)
inspektor-gadget/inspektor-gadget: docs/devel/hello-world-gadget.md

This macro will automatically create a ring buffer if the kernel supports it.
Otherwise, a perf array will be created.

> Optionally, you can employ the `GADGET_TRACER` macro to define the gadget as a tracer,
> specifying its behaviour with the following parameters:

Reviewer: I don't think we should say "define the gadget as a tracer" as very soon we'll support multiple tracers and even multiple kind of output in the same gadget.

```suggestion
Optionally, you can employ the `GADGET_TRACER` macro to define a tracer with the following parameters:
```

- Tracer's Name: `open`
- Buffer Map Name: `events`
- Event Structure Name: `event`

> This information enables Inspektor Gadget to generate the metadata file automatically.
> This file allows you to specify various aspects, including the output format of the gadget.
> Refer to the [metadata file](#creating-a-metadata-file) section for detailed instructions.
> However, it's important to note that this step is entirely optional. You can create the
> metadata file manually, or even not create it at all if it's unnecessary for your purposes.

Reviewer: Do we really need to be that verbose? What about the following?

```suggestion
This information enables Inspektor Gadget to generate the metadata file automatically.
Refer to the [metadata file](#creating-a-metadata-file) section for detailed instructions.
```

```c

> // [Optional] Define the gadget as a tracer

Reviewer: ```suggestion
// [Optional] Define a tracer
```

An initial version of the metadata file can be created by passing `--update-metadata` to the build command:

> > [!NOTE]
> > This is only valid if the eBPF program used one of the available macros to define the type of
> > the gadget (e.g. `GADGET_TRACER`). Check out the
> > [Implementing the eBPF program](#implementing-the-ebpf-program) section for more information.
> 

Reviewer: I'm not sure how to write it, but I'd like to avoid mentioning "type of the gadget" as this concept won't exist anymore with image-based gadgets.

Perhaps something like works?

```suggestion
> [!NOTE]
> This is only valid if the eBPF program used one the `GADGET_TRACER` macro to define a tracer. If such macro wasn't used, the metadata file has to be created by hand.

```

Author: Understood. I used:

```md
> [!NOTE]
> The `tracers` and `structs` sections will only be generated if the eBPF program defined a tracer
> using the `GADGET_TRACER` macro.
```
